benchexec/tools/btormc.py

```diff
@@ -0,0 +1,47 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.result as result
+import benchexec.tools.template
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for BtorMC -- A Bounded Model Checker for Btor2
+    """
+
+    REQUIRED_PATHS = ["build/"]
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("btormc", subdir="build/bin")
+
+    def name(self):
+        return "BtorMC"
+
+    def project_url(self):
+        return "https://github.com/Boolector/boolector"
+
+    def version(self, executable):
+        return self._version_from_tool(executable, stdin="")
+
+    def cmdline(self, executable, options, task, rlimits):
+        return [executable] + options + [task.single_input_file]
+
+    def determine_result(self, run):
+        """
+        @return: status of BtorMC after executing a run
+        """
+        if run.was_timeout:
+            return result.RESULT_TIMEOUT
+        for line in run.output[::-1]:
+            # BtorMC's option `--trace-gen` must be set to 1
+            # such that "sat/unsat" is printed to the output
```

Comment:
This option is set to 1 by BtorMC in its default mode,
but I could add it explicitly to `cmdline` as well.

Reply:
Ok, then I would tend to not enforce it. One could only consider checking it in `cmdline`, but even then I would say it is only a warning, not a failure, if the option is set to a different value. But likely other tools also have options that mangle the output in a way that the tool-info module would fail and we don't check for it.

Clarifying the comment that this is the default would be useful, though. Otherwise people like me think they have to do extra steps.

Reply:
Addressed in 78d4373